new_script.ps1

```diff
@@ -0,0 +1,172 @@
+#This script builds several branches for an Application in AppCenter and saves a report.
+#Check Readme.md file for more details - https://github.com/bumbazimba/appcenter_automatization/blob/master/README.md
+#Designed for Powershell 6 or above. Tested on Powershell 7
+#Author Nikita Nikolaev
+#Version 1.1
+
+#Enter AppCenter credentials for API requests
+
+[string]$user = Read-Host "Enter AppCenter username"
+[string]$app = Read-Host "Enter Application name"
+[string]$token = Read-Host "Enter API Token"
+[int]$branches_number = Read-Host "Enter number of branches to build"
+$branches_number = $branches_number-1
+
+
+$excel = Create-ExcelFile
+$branches = Get-Branches
+$build = Build-Branches $branches
+$build_details = Get-BuildDetails $build
+Insert-ExcelFile $excel, $build_details, $branches
+Save-ExcelFile $excel
+
+
+
+#Get All branches from the Application
+function Get-Branches ()
+{
+	$branches = Invoke-RestMethod -Uri "https://api.appcenter.ms/v0.1/apps/$user/$app/branches" -Method Get `
+		-Headers @{"Accept"="application/json"; "X-API-Token"="$($token)"; "Content-Type"="application/json"} -MaximumRetryCount 5 -RetryIntervalSec 2;
+	Write-Host "`nList of all branches in Application $app :"
+    Write-Host $branches
+    return $branches
+}
+
+
+#Build specified number of branches
+function Build-Branches ()
+{
+    param
+    (
+        [Parameter(Mandatory=$true)][object]$branches
+    )
+
+	$build = @()
+    
+	#Check if user wants to build All or just few branches
+	[int]$branches_count = $branches.Length - 1
+    $branches_names = $branches.branch.name
+        if ($branches_count -ne $branches_number)
+        {
+            $branches_count = $branches_number
+        }
+        else 
+        {
+            $branches_count = $branches_count - 1
+            $branches_number = $branches_count
+        }
+	Foreach ($branch in $branches_names[0..$branches_count])	
+		{ 		
+			$build += Invoke-RestMethod -Uri "https://api.appcenter.ms/v0.1/apps/$user/$app/branches/$branch/builds" -Method Post `
+				-Headers @{"Accept"="application/json"; "X-API-Token"="$($token)"; "Content-Type"="application/json"} `
+				-MaximumRetryCount 5 -RetryIntervalSec 2 | Select-object id
+			Write-Host "Branch $branch - build started"
+		}
+        return $build
+}
+
+#Get Details about all Builds
+function Get-BuildDetails ()
+{
+        param
+    (
+        [Parameter(Mandatory=$true)][object]$build
+    )
+	
+	$result = "in progress"
+	$number = 0
+    $build_result = @()
+		Foreach ($id in $build)
+		{
+			$id = $build[$number].id

```

Comment:
do we really need a `$number` here? and i believe `$id` is not really an id in `Foreach ($id in $build)`. can't we use something like:
```
Foreach ($build in $builds)
{
...$build.id...
}
```
And remove `$number` ?

Reply:
Thank you for comment, i was analyzing a bit.
I think number is needed, because in $build i actually several objects and i need to get an id of every object. Without $number, i am getting several ids instead of one. And i need each $id separately to perform API request.
This is the output of $build.id in cycle:
PS C:\WINDOWS\system32> $build.id
86
87
88

$build :

id              : 86
buildNumber     : 86
queueTime       : 2020-07-08T10:03:47.3185048Z
startTime       : 2020-07-08T10:04:03.9626571Z
finishTime      : 2020-07-08T10:06:45.6613123Z
lastChangedDate : 2020-07-08T10:06:46.167Z
status          : completed
result          : failed
reason          : manual
sourceBranch    : AdventureWorks_v1_camera_app
sourceVersion   : fc60d96ead84ae59c83a4b080b3c8eb77538307e
tags            : {normal, signed, manual, uwp}
properties      : 

id              : 87
buildNumber     : 87
...

If i need to get rid of $number, i will have to select string of the output.
Or, perhaps, there is a better way, which i don't know :)

Reply:
Got it. Thank you for checking this and for such a detailed explanation! 👍